api/prometheus.go

```diff
@@ -0,0 +1,908 @@
+		Name: "renterd_consensus_state_synced",
+		Value: func() float64 {
+			if c.Synced {
+				return 1
+			}
+			return 0
+		}(),
+	})
+	metrics = append(metrics, prometheus.Metric{
+		Name:  "renterd_consensus_state_chain_index_height",
+		Value: float64(c.BlockHeight),
+	})
+	return
+}
+func (asr AutopilotStateResponse) PrometheusMetric() (metrics []prometheus.Metric) {
+	labels := map[string]any{
+		"version":   asr.Version,
+		"commit":    asr.Commit,
+		"os":        asr.OS,
+		"buildTime": asr.BuildTime.String(),
+	}
+	// labels := fmt.Sprintf(`network="%s", version="%s", commit="%s", os="%s", buildTime="%s"`,
+	// build.NetworkName(), build.Version(), build.Commit(), runtime.GOOS, api.TimeRFC3339(build.BuildTime()))
+	metrics = append(metrics, prometheus.Metric{
+		Name:   "renterd_autopilot_state_uptimems",
+		Labels: labels,
+		Value:  float64(asr.UptimeMS),
+	})
+	metrics = append(metrics, prometheus.Metric{
+		Name:   "renterd_autopilot_state_configured",
+		Labels: labels,
+		Value: func() float64 {
+			if asr.Configured {
+				return 1
+			}
+			return 0
+		}(),
+	})
+	metrics = append(metrics, prometheus.Metric{
+		Name:   "renterd_autopilot_state_migrating",
+		Labels: labels,
+		Value: func() float64 {
+			if asr.Migrating {
+				return 1
+			}
+			return 0
+		}(),
+	})
+	metrics = append(metrics, prometheus.Metric{
+		Name:   "renterd_autopilot_state_migratinglaststart",
+		Labels: labels,
+		Value:  float64(time.Time(asr.MigratingLastStart).Unix()),
+	})
+	metrics = append(metrics, prometheus.Metric{
+		Name:   "renterd_autopilot_state_pruning",
+		Labels: labels,
+		Value: func() float64 {
+			if asr.Pruning {
+				return 1
+			}
+			return 0
+		}(),
+	})
+	metrics = append(metrics, prometheus.Metric{
+		Name:   "renterd_autopilot_state_pruninglaststart",
+		Labels: labels,
+		Value:  float64(time.Time(asr.PruningLastStart).Unix()),
+	})
+	metrics = append(metrics, prometheus.Metric{
+		Name:   "renterd_autopilot_state_scanning",
+		Labels: labels,
+		Value: func() float64 {
+			if asr.Scanning {
+				return 1
+			}
+			return 0
+		}(),
+	})
+	metrics = append(metrics, prometheus.Metric{
+		Name:   "renterd_autopilot_state_scanninglaststart",
+		Labels: labels,
+		Value:  float64(time.Time(asr.ScanningLastStart).Unix()),
+	})
+	return
+}
+
+func (b Bucket) PrometheusMetric() (metrics []prometheus.Metric) {
+	metrics = append(metrics, prometheus.Metric{
+		Name: "renterd_bucket",
+		Labels: map[string]any{
+			"name": b.Name,
+			"publicReadAccess": func() float64 {
+				if b.Policy.PublicReadAccess {
+					return 1
+				}
+				return 0
+			}(),
+			"createdAt": b.CreatedAt.String(),
+		},
+		Value: 1,
+	})
+	return
+}
+
+func (host Host) PrometheusMetric() (metrics []prometheus.Metric) {
+	priceTableLabels := map[string]any{
+		"netAddress": host.NetAddress,
+		"uid":        host.PriceTable.UID.String(),
+		"expiry":     host.PriceTable.Expiry.Local().Format("2006-01-02T15:04:05Z07:00"),
```

Comment:
`nit` use `.Format(time.RFC3339)` so if we ever change time formats a full search also shows this line